src/main/cpp/src/vcf/htslib_fs_adapter.cc

```diff
@@ -0,0 +1,88 @@
+/**
+ * The MIT License (MIT)
+ * Copyright (c) 2020 Omics Data Automation, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "hfile_genomicsdb.h"
+
+#include "tiledb.h"
+#include "tiledb_storage.h"
+
+#include <mutex>
+
+std::once_flag initialize_once;
+
+void genomicsdb_htslib_plugin_initialize(const char* filename) {
+  if (strstr(filename, "://")) {
+    std::call_once(initialize_once, [&](){
```

Comment:
My preference - I don't like lambda functions. Function names are quite useful for reading code.
In this case, it's fine since it's only 2 lines.

Reply:
Agreed, this was just a simple function.